File: Tyche.iOS/Pool/Sources/Pool/UI/PoolGamblerScoreModel.swift
struct PoolGamblerScoreModelId: Hashable, Codable {
    let poolId: String
    let gamblerId: String
}

struct PoolGamblerScoreModel: Identifiable, Hashable, Codable {
    let poolId: String
    let poolName: String
    let gamblerId: String
    let gamblerUsername: String
    let currentPosition: Int?
    let beforePosition: Int?
    let score: Int?
    
    var id: PoolGamblerScoreModelId {
        return PoolGamblerScoreModelId(poolId: poolId, gamblerId: gamblerId)
    }
}

extension PoolGamblerScoreModel {
    func calculateDifference() -> Int? {
        guard let currentPosition = self.currentPosition else {
            return nil
        }
        
        guard let beforePosition = self.beforePosition else {
            return nil
        }
        
        return beforePosition - currentPosition
    }
}

File: Tyche.Android/pool/src/main/java/com/felipearpa/tyche/pool/PoolGamblerScoreModel.kt
package com.felipearpa.tyche.pool

data class PoolGamblerScoreModel(
    val poolId: String,
    val poolName: String,
    val gamblerId: String,
    val gamblerUsername: String,
    val currentPosition: Int?,
    val beforePosition: Int?,
    val score: Int?
) {

    fun difference(): Int? {
        currentPosition?.let { currentPosition ->
            beforePosition?.let { bp ->
                return bp - currentPosition
            }
            return currentPosition
        }
        return null
    }
}